### PR TITLE
HSEARCH-4002 + HSEARCH-4003 + HSEARCH-4004 MappingAnnotatedElement improvements

### PR DESCRIPTION
**Pull request by yrodiere**
* [HSEARCH-4003](https://hibernate.atlassian.net/browse/HSEARCH-4003): Automatically expand repeatable annotations in MappingAnnotatedElement
* [HSEARCH-4002](https://hibernate.atlassian.net/browse/HSEARCH-4002): Expose the type of a property for a given ContainerExtractorPath in MappingAnnotatedProperty
* [HSEARCH-4004](https://hibernate.atlassian.net/browse/HSEARCH-4004): Expose an eventContext from MappingAnnotationProcessorContext

Necessary for the Search 5 compatibility layer.

### CODE REVIEW

**Comment by yrodiere** [comment]
Jenkins, retest this please

File: integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/annotation/processing/CustomPropertyMappingAnnotationBaseIT.java [hunk 1]
```diff
@@ -228,4 +270,66 @@ else if ( Integer.class.equals( annotatedElement.javaClass() ) ) {
...
+
+		@Retention(RetentionPolicy.RUNTIME)
+		@Target({ElementType.FIELD, ElementType.METHOD})
+		@interface List {
+			MultiFieldAnnotation[] value();
+		}
+
+	}
+
+
```
Comment (fax4ever) [inline integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/annotation/processing/CustomPropertyMappingAnnotationBaseIT.java:289]: extra space here

**Review by fax4ever — COMMENTED** [review]
Nice job. Just 2 not critical comments.

File: mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/processing/MappingAnnotatedProperty.java [hunk 2]
```diff
@@ -14,6 +18,19 @@
...
+	 */
 	String name();
 
+	/**
+	 * @param extractorPath A container extractor path, possibly just {@link ContainerExtractorPath#defaultExtractors()}.
+	 * @return The Java class corresponding to the raw type of values
+	 * extracted from this property with the given container extractor path,
+	 * or {@link Optional#empty()} if the path cannot be applied.
+	 */
+	Optional<Class<?>> extractedValueJavaClass(ContainerExtractorPath extractorPath);
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/processing/MappingAnnotatedProperty.java:34]: Right. Renamed to `javaClass(ContainerExtractorPath)`.

[the remark this answers is not in the document]
Comment (yrodiere) [reply mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/processing/MappingAnnotatedProperty.java:34]: And added a test.

**Comment by yrodiere** [comment]
Thanks. Addresssed your comments and force-pushed.

**Comment by yrodiere** [comment]
Merged, thanks

**Pull request merged**